### PR TITLE
Use 2 separate pools to fix a problem in ParquetReaderBenchmark.cpp

### PR DESCRIPTION
**Pull request by yiweiHeOSS**
**Problem description:**
Tried to run the ParquetReaderBenchmark(velox_dwio_parquet_reader_benchmark) and got below error:
```
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Breakpoint 1, facebook::velox::detail::veloxCheckFail<facebook::velox::VeloxRuntimeError, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&> (args=..., s="Memory pool addAggregateChild operation is only allowed on aggregation memory pool: Memory Pool[default_leaf_0 LEAF root[__default_root__] parent[__default_root__] MALLOC no-usage-track thread-safe]<u"...) at /root/velox/./velox/common/base/Exceptions.h:69
69	    LOG(ERROR) << "Line: " << args.file << ":" << args.line
E0811 15:19:40.412384 481990 Exceptions.h:69] Line: /root/velox/velox/common/memory/MemoryPool.cpp:313, Function:addAggregateChild, Expression:  Memory pool addAggregateChild operation is only allowed on aggregation memory pool: Memory Pool[default_leaf_0 LEAF root[__default_root__] parent[__default_root__] MALLOC no-usage-track thread-safe]<unlimited max capacity unlimited capacity used 0B available 0B reservation [used 0B, reserved 0B, min 0B] counters [allocs 0, frees 0, reserves 0, releases 0, collisions 0])>, Source: RUNTIME, ErrorCode: INVALID_STATE
terminate called after throwing an instance of 'facebook::velox::VeloxRuntimeError'
  what():  Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: Memory pool addAggregateChild operation is only allowed on aggregation memory pool: Memory Pool[default_leaf_0 LEAF root[__default_root__] parent[__default_root__] MALLOC no-usage-track thread-safe]<unlimited max capacity unlimited capacity used 0B available 0B reservation [used 0B, reserved 0B, min 0B] counters [allocs 0, frees 0, reserves 0, releases 0, collisions 0])>
Retriable: False
Function: addAggregateChild
File: /root/velox/velox/common/memory/MemoryPool.cpp
Line: 313
...
Program
received signal SIGABRT, Aborted.
__pthread_kill_implementation (no_tid=0, signo=6, threadid=140737309463040) at ./nptl/pthread_kill.c:44
```
Stack:
```
facebook::velox::detail::veloxCheckFail<facebook::velox::VeloxRuntimeError, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&> (\root\velox\velox\common\base\Exceptions.h:69)
facebook::velox::ram2::MemoryPool::addAggregateChild (\root\velox\velox\common\memory\MemoryPool.cpp:313)
facebook::velox::parquet::writer::Writer (\root\velox\velox\dwio\parquet\writer\Writer.cpp:137)
std::make_unique<facebook::velox::parquet::Writer, std::unique_ptr<facebook::velox::dwio::common::WriteFileDataSink, std::default_delete<facebook::velox::dwio::common::WriteFileDataSink> >, facebook::velox::parquet::WriterOptions&> (\usr\include\c++\11\bits\unique_ptr.h:962)
ParquetReaderBenchmark::ParquetReaderBenchmark (\root\velox\velox\dwio\parquet\tests\reader\ParquetReaderBenchmark.cpp:56)
run (\root\velox\velox\dwio\parquet\tests\reader\ParquetReaderBenchmark.cpp:251)
run_BigInt_Filter_0_Nulls_0_next_5k_dict (\root\velox\velox\dwio\parquet\tests\reader\ParquetReaderBenchmark.cpp:369)
operator() (\root\velox\velox\dwio\parquet\tests\reader\ParquetReaderBenchmark.cpp:369)
operator() (\usr\local\include\folly\Benchmark.h:204)
std::__invoke_impl<folly::detail::TimeIterData, folly::addBenchmark<<lambda(unsigned int)> >(char const*, folly::StringPiece, <lambda(unsigned int)>&&)::<lambda(unsigned int)>&, unsigned int>(std::__invoke_other, struct {...} &) (\usr\include\c++\11\bits\invoke.h:61)
std::__invoke_r<folly::detail::TimeIterData, folly::addBenchmark<<lambda(unsigned int)> >(char const*, folly::StringPiece, <lambda(unsigned int)>&&)::<lambda(unsigned int)>&, unsigned int>(struct {...} &) (\usr\include\c++\11\bits\invoke.h:116)
std::_Function_handler<folly::detail::TimeIterData(unsigned int), folly::addBenchmark<<lambda(unsigned int)> >(char const*, folly::StringPiece, <lambda(unsigned int)>&&)::<lambda(unsigned int)> >::_M_invoke(const std::_Any_data &, unsigned int &&) (\usr\include\c++\11\bits\std_function.h:291)
std::function<folly::detail::TimeIterData (unsigned int)>::operator()(unsigned int) const (\usr\include\c++\11\bits\std_function.h:590)
folly::runBenchmarkGetNSPerIteration (\root\velox\folly\folly\Benchmark.cpp:169)
folly::(anonymous namespace)::runBenchmarksWithPrinter (\root\velox\folly\folly\Benchmark.cpp:682)
folly::runBenchmarks (\root\velox\folly\folly\Benchmark.cpp:756)
main (\root\velox\velox\dwio\parquet\tests\reader\ParquetReaderBenchmark.cpp:378)
```

**Solution:**

I see there are only 2 kinds of memory pool:
```
/// Defines the kinds of a memory pool.
  enum class Kind {
    /// The leaf memory pool is used for memory allocation. User can allocate
    /// memory from this kind of pool but can't create child pool from it.
    kLeaf = 0,
    /// The aggregation memory pool is used to manage the memory pool hierarchy
    /// and aggregate the memory usage from the leaf pools. The user can't
    /// directly allocate memory from this kind of pool but can create child
    /// pools from it.
    kAggregate = 1,
  };
```
so I changed the code in ParquetReaderBenchmark.cpp to use 2 separate pools and fixed that problem (follow the code in E2EFilterTest.cpp):
```
    rootPool_ = memory::defaultMemoryManager().addRootPool("ParquetReaderBenchmark");
    leafPool_ = rootPool_->addLeafChild("ParquetReaderBenchmark"); 
```

### CODE REVIEW

File: velox/dwio/parquet/tests/reader/ParquetReaderBenchmark.cpp [hunk 1]
```diff
@@ -41,8 +41,10 @@ class ParquetReaderBenchmark {
  public:
   explicit ParquetReaderBenchmark(bool disableDictionary)
       : disableDictionary_(disableDictionary) {
-    pool_ = memory::addDefaultLeafMemoryPool();
-    dataSetBuilder_ = std::make_unique<DataSetBuilder>(*pool_.get(), 0);
+    rootPool_ =
+        memory::defaultMemoryManager().addRootPool("ParquetReaderBenchmark");
+    leafPool_ = rootPool_->addLeafChild("ParquetReaderBenchmark");
+    dataSetBuilder_ = std::make_unique<DataSetBuilder>(*leafPool_.get(), 0);
```
[the remark this answers is not in the document]
Comment (yiweiHeOSS) [reply velox/dwio/parquet/tests/reader/ParquetReaderBenchmark.cpp]: Ok, fixed it.

**Comment by yiweiHeOSS** [comment]
This PR is part of work to solve https://github.com/facebookincubator/velox/issues/6070

**Review by yingsu00 — COMMENTED** [review]
@yiweiHeOSS Can you please squash the commits into 1?

**Comment by yiweiHeOSS** [comment]
@yingsu00 Ok, done. I 'rebase commits' and 'push force' to this PR.

**Review by yingsu00 — APPROVED** [review]

**Comment by yingsu00** [comment]
@Yuhta Will you be able to review this PR? Thanks!

**Comment by Yuhta** [comment]
This should already have been merged in a77b987c81c4d0f9b5f8b02d614a481c541c5b58, not sure why the PR is not closed automatically

**Comment by yingsu00** [comment]
Closing as it's already merged a77b987c8

**Pull request closed without merging**